packages/headless/src/integration-tests/field-suggestions.test.ts

```diff
@@ -0,0 +1,253 @@
+import {
+  buildSearchEngine,
+  getSampleSearchEngineConfiguration,
+  SearchEngine,
+  buildFieldSuggestions,
+  FieldSuggestions,
+  Controller,
+  buildFacet,
+  Facet,
+  FieldSuggestionsValue,
+  CoreEngine,
+} from '..';
+
+function isEngine(obj: object): obj is CoreEngine {
+  return 'dispatch' in obj;
+}
+
+function waitForNextStateChange(
+  target: Controller | CoreEngine,
+  options: {action?: () => void; expectedSubscriberCalls?: number} = {}
+) {
+  return new Promise<void>((resolve) => {
+    let skipFirstCall = !isEngine(target);
+    let subscriberCalls = 0;
+    let unsubscribe = () => {};
+    unsubscribe = target.subscribe(() => {
+      if (skipFirstCall) {
+        skipFirstCall = false;
+        return;
+      }
+      if (++subscriberCalls < (options?.expectedSubscriberCalls ?? 1)) {
```

Comment:
Instead of doing like Cypress and waiting for the desired state, I thought it would be preferable to wait for a certain number of listener calls, which I think is more fitting for functional tests.